website_portal_contract/views/website_portal_contract_templates.xml

```diff
@@ -0,0 +1,350 @@
+                                    contract debit, credit, and balance.
+                                </p>
+                                <table class="table wq-table">
+                                    <thead>
+                                        <tr>
+                                            <th>Name</th>
+                                            <th>Quantity</th>
+                                            <th>UoM</th>
+                                            <th>Subtotal</th>
+                                            <th class="text-right">Debit</th>
+                                            <th class="text-right">Credit</th>
+                                            <th class="text-right">Balance</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        <tr>
+                                            <td>
+                                                <span>Development Services</span>
+                                            </td>
+                                            <td>
+                                                <span>100.0</span>
+                                            </td>
+                                            <td>
+                                                <span>Hour(s)</span>
+                                            </td>
+                                            <td>
+                                                <span>8000.0</span>
+                                            </td>
+                                        </tr>
+                                        <tr>
+                                            <td /><td /><td /><td />
+                                            <td class="text-right">
+                                                <span>8000.0</span>
+                                            </td>
+                                            <td class="text-right">
+                                                <span>8000.0</span>
+                                            </td>
+                                            <td class="text-right">
+                                                <strong><span>8000.0</span></strong>
+                                            </td>
+                                        </tr>
+                                    </tbody>
+                                </table>
+                            </section>
+                            <div id="template_introduction" t-field="template.website_description" class="oe_no_empty" />
+                        </div>
+                    </div>
+                </div>
+            </body>
+        </t>
+    </template>
+
+    <template id="contract_invoices" name="Invoices">
+        <section id="invoice_and_balance">
+            <h1 class="page-header">Invoices &amp; Balance</h1>
```

Comment:
You're showing the recurring invoice lines here, not the actual invoices that were generated. It would be best to make one section "Recurring Items", then another "Invoices & Balance", which iterates the invoices themselves.

In order to get the invoices, you can use the [`contract_show_invoice`](https://github.com/OCA/contract/blob/10.0/contract_show_invoice/models/invoice.py) module & just search for the contract ID on the invoice model in your controller.